src/probnum/diffeq/odesolution.py

```diff
@@ -181,3 +189,25 @@ def __getitem__(self, idx):
             return _RandomVariableList(f_rvs)
         else:
             raise ValueError("Invalid index")
+
+    def sample(self, t=None, size=()):
+        # this has its own recursion because of the tedious undoing of preconditioning....
+
+        size = utils.as_shape(size)
+
+        # implement only single samples, rest via recursion
+        if size != ():
+            return np.array([self.sample(t=t, size=size[1:]) for _ in range(size[0])])
+
+        samples = self._kalman_posterior.sample(locations=t, size=size)
+        return self._undo_preconditioning_and_project_samples(samples)
+
+    def _undo_preconditioning_and_project_samples(self, samples):
```

Comment:
This kind of relates to my comment above, so just a reminder to adapt this part depending on what you end up changing above.

Reply:
Maybe you could also have a `_undo_preconditioning_and_project_rv`, and just call
```python
_undo_preconditioning_and_project_rv([asrandvar(s) for s in samples]).mean
```
Again, no idea what of all this would be best